### PR TITLE
[Term Entry] Java ArrayList .removeAll()

### PR DESCRIPTION
**Pull request by bpicoCode**
### Description

- Adding a new term entry for ArrayList .removeAll() method

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+A collection of elements, specified by `Collection c`, can be removed from an `ArrayList` instance using the `.removeAll()` method:
+
+```pseudo
+arrayListInstance.removeAll(Collection c);
+```
+
+## Example
+
+In the example below, two `ArrayList` instance `fallMonths` and `monthsToRemove` are created and can hold `String`-type elements. Next, elements are added with [the `.add()` method](https://www.codecademy.com/resources/docs/java/array-list/add). Then, `monthsToRemove` is utilized to remove the months August and September from `fallMonths` using `.removeAll()`. Finally, all elements are removed from `fallMonths` by invoking `.removeAll()` with a reference to itself.
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:27]:
```suggestion
In the example below, two `ArrayList` instances, `fallMonths` and `monthsToRemove`, are created to hold `String`-type elements. Elements are added to the `fallMonths` using the [`.add()`] method (https://www.codecademy.com/resources/docs/java/array-list/add). Then, `monthsToRemove` is utilized to remove the months - August and September from `fallMonths` using `.removeAll()`. Finally, all elements are removed from `fallMonths` by invoking `.removeAll()` with a reference to itself.
```

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    ArrayList<String> monthsToRemove = new ArrayList<String>();
+    
+    monthsToRemove.add("August");
+    monthsToRemove.add("December");
+
+    // Remove August and December from fallMonths using monthsToRemove
+    fallMonths.removeAll(monthsToRemove);
+
+    System.out.println(fallMonths);
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:51]:
```suggestion
    System.out.println("After removing specified months: " + fallMonths);```

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    
+    // Remove all elements from fallMonths
+    fallMonths.removeAll(fallMonths);
+
+    System.out.println(fallMonths);
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:56]:
```suggestion
System.out.println("After removing all elements: " + fallMonths);```

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The `.removeAll()` method is used for removing multiple elements from instances of the `ArrayList` class.
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:15]:
```suggestion
The **`.removeAll()`** method is used for removing multiple elements from instances of the `ArrayList` class.
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @bpicoCode!
Thank you for contributing to Codecademy Docs! The entry is well written!

I've suggested a few changes, could you please review and modify them at your earliest convenience? 
Thank you!

**Comment by bpicoCode** [comment]
@mamtawardhani Suggestions committed!

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+A collection of elements, specified by `Collection c`, can be removed from an `ArrayList` instance using the `.removeAll()` method:
+
+```pseudo
+arrayListInstance.removeAll(Collection c);
+```
+
+## Example
+
+In the example below, two `ArrayList` instances, `fallMonths` and `monthsToRemove`, are created to hold `String`-type elements. Elements are added to the `fallMonths` using the [`.add()`] method (https://www.codecademy.com/resources/docs/java/array-list/add). Then, `monthsToRemove` is utilized to remove the months - August and September from `fallMonths` using `.removeAll()`. Finally, all elements are removed from `fallMonths` by invoking `.removeAll()` with a reference to itself.
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:28]:
```suggestion
In the example below, two `ArrayList` instances, `fallMonths` and `monthsToRemove`, are created to hold `String`-type elements. Elements are added to the `fallMonths` using the [`.add()`]
(https://www.codecademy.com/resources/docs/java/array-list/add) method. Then, `monthsToRemove` is utilized to remove the months - August and September from `fallMonths` using `.removeAll()`. Finally, all elements are removed from `fallMonths` by invoking `.removeAll()` with a reference to itself.
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @bpicoCode, just one change is required, please review and modify it!
Thanks!

**Review by mamtawardhani — APPROVED** [review]
Good for the second review

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.removeAll()`** method is used for removing multiple elements from instances of the `ArrayList` class.
+
+## Syntax
+
+A collection of elements, specified by `Collection c`, can be removed from an `ArrayList` instance using the `.removeAll()` method:
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:20]: Remove this line and add the description of the parameters `ArrayList2` and `ArrayList1` below the syntax in this format - 
```
- `parameterName`: Parameter description.
- `parameterName`: Parameter description.
```

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+```pseudo
+arrayListInstance.removeAll(Collection c);
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:23]:
```suggestion
ArrayList1.removeAll(ArrayList2);
```
Simply the syntax as much as possible.

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    fallMonths.removeAll(monthsToRemove);
+
+    System.out.println("After removing specified months: " + fallMonths);```
+    
+    // Remove all elements from fallMonths
+    fallMonths.removeAll(fallMonths);
+
+System.out.println("After removing all elements: " + fallMonths);```
+  }
+}
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:60]:
```suggestion
import java.util.ArrayList;

public class RemoveFallMonths{
  public static void main(String[] args) {

    ArrayList<String> fallMonths = new ArrayList<String>();

    fallMonths.add("August");
    fallMonths.add("September");
    fallMonths.add("October");
    fallMonths.add("November");
    fallMonths.add("December");

    ArrayList<String> monthsToRemove = new ArrayList<String>();

    monthsToRemove.add("August");
    monthsToRemove.add("December");

    // Remove August and December from fallMonths using monthsToRemove
    fallMonths.removeAll(monthsToRemove);

    System.out.println("After removing specified months: " + fallMonths);

    // Remove all elements from fallMonths
    fallMonths.removeAll(fallMonths);

System.out.println("After removing all elements: " + fallMonths);
  }
}
```
The code had some errors, but I fixed them.

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+The `RemoveFallMonths` output should look like this:
+
+```shell
+[September, October, November]
+[]
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:67]:
```suggestion
After removing specified months: [September, October, November]
After removing all elements: []
```
This is the complete output.

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    
+    // Remove all elements from fallMonths
+    fallMonths.removeAll(fallMonths);
+
+System.out.println("After removing all elements: " + fallMonths);```
+  }
+}
+```
+
+The `RemoveFallMonths` output should look like this:
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:63]:
```suggestion
The output of the above code should look like this:
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @bpicoCode, I suggested some changes, please make them.

**Comment by avdhoottt** [comment]
Hey @bpicoCode, it's been two weeks since I suggested the changes, are you still working on this PR?

File: content/java/concepts/array-list/terms/removeAll/removeAll.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+```pseudo
+  ArrayList1.removeAll(ArrayList2);
+```
+
+- `ArrayList1`: It is the `ArrayList` from which elements will be removed.
+- `ArrayList2`: It is the `ArrayList` containing the elements to be removed from `ArrayList1`.
+
+## Example
+
+In the example below, two `ArrayList` instances, `fallMonths` and `monthsToRemove`, are created to hold `String`-type elements. Elements are added to the `fallMonths` using the [`.add()`]
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/removeAll/removeAll.md:29]:
```suggestion
In the example below, two `ArrayList` instances, `fallMonths` and `monthsToRemove`, are created to hold `String` type elements. Elements are added to the `fallMonths` using the [`.add()`]
```

**Review by avdhoottt — APPROVED** [review]
LGTM!!

**Pull request merged**